test/resources/resources.cc

```diff
@@ -207,13 +240,14 @@ Variant<AssertionResult, std::string> get_transformed_ir(
 AssertionResult test_pretty_printer(const std::string& filename) {
   if (ends_with(filename, ".ref")) return AssertionSuccess();
   assert(ends_with(filename, ".gh"));
-  std::string ref_filename =
-      filename.substr(0, filename.size() - sizeof(".gh") + 1) + ".ref";
   auto first_pass = get_pretty_printed_file(filename);
   if (first_pass.is<AssertionResult>())
     return first_pass.get_unchecked<AssertionResult>();
   const std::string& first_pass_result =
       first_pass.get_unchecked<std::string>();
+
+  std::string ref_filename =
+      filename.substr(0, filename.size() - sizeof(".gh") + 1) + ".ref";
```

Comment:
Also if you want to keep that kind of operation, make a function that would be reusable